hoomd/operations.py

```diff
@@ -230,12 +230,12 @@ def __contains__(self, operation):
     def __iter__(self):
         """Iterates through all contained operations."""
         integrator = (self._integrator,) if self._integrator else []
-        yield from chain(self._tuners, self._updaters, integrator,
-                         self._writers, self._computes)
+        yield from chain(integrator, self._updaters, self._writers,
+                         self._tuners, self._computes)
 
     def __len__(self):
         """Return the number of operations contained in this collection."""
-        return len(list(self))
+        return len([None for _ in self])
```

Comment:
Why not add the lengths of the contained lists (and add 1 if there is an integrator)?

Reply:
I didn't do this to keep the code somewhat simpler, (utilizing the existing `__iter__`), but there would be nothing wrong with adding list lengths.